### PR TITLE
Local changes

### PR DESCRIPTION
**Pull request by joaobrittoneto**
Make Task component port_driven. Add unit test for a simple terminal interface.

### CODE REVIEW

**Comment by doudou** [comment]
*Please* do not call your pull requests "local changes"

They're all "local changes"

File: test/test_write_data.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        sample = data_dock.raw_data_input.new_sample
+        sample.time = Time.now
+        sample
+    end
+
+    it 'input written data' do
+        data_dock.configure
+        data_dock.start
+
+        print "Enter data to auv: "
```
Comment (doudou) [inline test/test_write_data.rb]: Don't require tests to have a human behind them. The human does not know what is a good "data test". You do.

**Comment by doudou** [comment]
Also: split your commits in logical blocks. Git gui allows you to choose what to commit and what to not commit line-by-line.

For instance, here, it's critical that you explain why you remove the HUGE_DATA_SIZE exception state. It's very unclear to me, for once.

Also, split your commits between the summary and the full description

```
remove exception state FULL_PACKET_QUEUE

Blablablabla explain why
```

```
remove unused exception state FULL_PACKET_QUEUE

# If it was removed in a previous commit, mention it
# Otherwise, just don't add any extra description
```

**Comment by joaobrittoneto** [comment]
Sorry, I fix it.

About the HUGE_DATA_SIZE exception: 
In the unit test (sending raw_data) it goes to exception first in the library, when it tries to mount the packet. I realize now it's better to keep the exception in the task (In case the packet is mounted somewhere else).

**Pull request merged**